backend/fms_core/models/dataset.py

```diff
@@ -0,0 +1,36 @@
+from django.db import models
+from django.core.exceptions import ValidationError
+import reversion
+
+from .tracked_model import TrackedModel
+from .project import Project
+from .experiment_run import ExperimentRun
+from .container import Container
+
+from ._constants import STANDARD_NAME_FIELD_LENGTH
+
+@reversion.register()
+class Dataset(TrackedModel):
+    """ Class to store information about the datasets of data deliveries. """
+
+    project_name = models.CharField(max_length=STANDARD_NAME_FIELD_LENGTH, help_text="Project name.")
+
+    run_name = models.CharField(max_length=STANDARD_NAME_FIELD_LENGTH, help_text="Run name.")
+
+    lane = models.CharField(max_length=10, blank=True, help_text="Coordinates of the lane in a container")
+
+    def save(self, *args, **kwargs):
+        errors = {}
+
+        try:
+            int(self.lane)
+        except ValueError as e:
+            errors["ValueError"] = "The lane must be a positive integer"
+        
+        if Dataset.objects.filter(project_name=self.project_name, run_name=self.run_name, lane=self.lane).exists():
```

Comment:
This is a clean way to prevent adding existing duplicates using the app, but it does not prevent the injection of erroneous row in the DB using SQL. It would still be safer to include a constraint to make sure we do not make an error during a DB curation. It also does not prevent the creation of Dataset that have different case for project or run, but are otherwise identical.

Reply:
> It would still be safer to include a constraint to make sure we do not make an error during a DB curation.
Which constraint am I missing?